Tidy up of the ft5406 driver to use DT

JamesH65 opened the pull request:
Driver was using a fixed resolution, this commit
adds coordinate size, flip and swap features
via device tree.

Adds overrides so the VC4 can adjust the DT parameters
appropriately, there is a newer version of the VC4 side
driver that can now set up the appropriate DT values
if requried.

Commit:
Tidy up of the ft5406 driver to use DT

Driver was using a fixed resolution, this commit
adds touchscreen size, and coordinate flip and swap
features via device tree overlays.

Adds overrides so the VC4 can adjust the DT parameters
appropriately; there is a newer version of the VC4 side
driver that can now set up the appropriate DT values
if requried.

Signed-off-by: James Hughes <james.hughes@raspberrypi.org>

pelwell commented on drivers/input/touchscreen/rpi-ft5406.c:
```diff
@@ -202,15 +222,37 @@ static int ft5406_probe(struct platform_device *pdev)
+	if (of_property_read_u32(np, "touchscreen-inverted-y", &val) >= 0)
+		ts->vflip = val;
+
+	if (of_property_read_u32(np, "touchscreen-swapped-x-y", &val) >= 0) {
```
You missed one.

pelwell commented on arch/arm/boot/dts/overlays/rpi-ft5406-overlay.dts:
```diff
@@ -11,7 +11,20 @@
+				touchscreen-size-x = <800>;
+				touchscreen-size-y = <600>;
+				touchscreen-inverted-x = <0>;
+				touchscreen-inverted-y = <0>;	
```
git show is highlighting stray whitespace at the end of this line.

pelwell commented on drivers/input/touchscreen/rpi-ft5406.c:
```diff
@@ -202,15 +222,36 @@ static int ft5406_probe(struct platform_device *pdev)
 
 	input_set_abs_params(ts->input_dev, ABS_MT_POSITION_X, 0,
-			     SCREEN_WIDTH, 0, 0);
+				ts->xyswap ? ts->max_y : ts->max_x, 0, 0);
```
The indentation on this and the next statement doesn't match the Linux style, which would use 3 TABs and 5 spaces.

pelwell reviewed (commented):
GitHub could do a better job with whitespace...

JamesH65 commented:
I've 'kernelified' this driver to coding standards (apart from a couple of warnings).

I think this can be merged without the matching firmware change since any flips previously done automatically by the firmware side can now be achieved by using the device tree settings.

pelwell commented:
Looks good to me.

JamesH65 commented:
@popcornmix @pelwell @6by9 

Now updated to top of tree and can now be merged, if reviews OK.

pelwell commented:
Sorry James - I've just spotted that you need to document the parameters in the README. And your commit message has a few typos. Otherwise it still looks good.

6by9 commented on drivers/input/touchscreen/rpi-ft5406.c:
```diff
@@ -164,17 +212,23 @@ static int ft5406_probe(struct platform_device *pdev)
-				    &touchbuf, sizeof(touchbuf));
+		dev_warn(dev,
+			 "set failed, trying get"
+			" (err:%d touchbuf:%x virt:%p bus:%x)\n",
```
Checkpatch should have complained at that - strings presented to the user shouldn't be broken even if it does take the line over 80 chars.

6by9 reviewed (commented):
One very minor nitpick, otherwise looks fine.

The driver ought to be updated to comply with the kernel coding style though (Mainly no C99 // commenting, and multi-line comments)

JamesH65 commented:
@pelwell @6by9 I have updated so able to merge, and updated the readme/comments as requested.

pelwell committed:
SQUASH: Include parameter usage in example

This keeps the checker happy.

pelwell committed:
SQUASH: Demote probe deferral error to an info

pelwell committed:
SQUASH: Remove deferral message altogether

Driver probe deferral is a normal part of Linux life, so don't generate any
confusing warnings when it happens.
